File: files/CH-191.R
library(tidyverse)
library(readxl)

path = "files/CH-191Column Splitting.xlsx"
input = read_excel(path, range = "B2:B7")
test  = read_excel(path, range = "D2:F7")

result = input %>% 
  extract(ID, into = c("Part 1", "Separator", "Part 2"), "([A-Za-z0-9]+)([^A-Za-z0-9])(.*)")
all.equal(result, test, check.attributes = FALSE)
# [1] TRUE

File: files/100-199/191/CH-191.py
import pandas as pd
import re

path = "CH-191Column Splitting.xlsx"

input = pd.read_excel(path, usecols="B", skiprows=1, nrows=6)
test = pd.read_excel(path, usecols="D:F", skiprows=1, nrows=6).astype(str)

input[['Part 1', 'Separator', 'Part 2']] = input.iloc[:, 0].str.extract(r"([A-Za-z0-9]+)([^A-Za-z0-9])(.*)").astype(str)
result = input[['Part 1', 'Separator', 'Part 2']]
print(result.equals(test))  # True